lib/ipam/ipam.go

```diff
@@ -33,80 +35,18 @@ const (
-	ReleaseIPs(ips []net.IP) ([]net.IP, error)
-
-	// GetAssignmentAttributes returns the attributes stored with the given IP address
-	// upon assignment.
-	GetAssignmentAttributes(addr net.IP) (map[string]string, error)
-
-	// IpsByHandle returns a list of all IP addresses that have been
-	// assigned using the provided handle.
-	IPsByHandle(handleID string) ([]net.IP, error)
-
-	// ReleaseByHandle releases all IP addresses that have been assigned
-	// using the provided handle.  Returns an error if no addresses
-	// are assigned with the given handle.
-	ReleaseByHandle(handleID string) error
-
-	// ClaimAffinity claims affinity to the given host for all blocks
-	// within the given CIDR.  The given CIDR must fall within a configured
-	// pool. If an empty string is passed as the host, then the value returned by os.Hostname is used.
-	ClaimAffinity(cidr net.IPNet, host string) ([]net.IPNet, []net.IPNet, error)
-
-	// ReleaseAffinity releases affinity for all blocks within the given CIDR
-	// on the given host.  If an empty string is passed as the host, then the
-	// value returned by os.Hostname will be used.
-	ReleaseAffinity(cidr net.IPNet, host string) error
-
-	// ReleaseHostAffinities releases affinity for all blocks that are affine
-	// to the given host.  If an empty string is passed as the host, the value returned by
-	// os.Hostname will be used.
-	ReleaseHostAffinities(host string) error
-
-	// ReleasePoolAffinities releases affinity for all blocks within
-	// the specified pool across all hosts.
-	ReleasePoolAffinities(pool net.IPNet) error
-
-	// GetIPAMConfig returns the global IPAM configuration.  If no IPAM configuration
-	// has been set, returns a default configuration with StrictAffinity disabled
-	// and AutoAllocateBlocks enabled.
-	GetIPAMConfig() (*IPAMConfig, error)
-
-	// SetIPAMConfig sets global IPAM configuration.  This can only
-	// be done when there are no allocated blocks and IP addresses.
-	SetIPAMConfig(cfg IPAMConfig) error
-
-	// RemoveIPAMHost releases affinity for all blocks on the given host,
-	// and removes all host-specific IPAM data from the datastore.
-	// RemoveIPAMHost does not release any IP addresses claimed on the given host.
-	// If an empty string is passed as the host then the value returned by os.Hostname is used.
-	RemoveIPAMHost(host string) error
-}
-
-// newIPAM returns a new ipamClient, which implements the IPAMInterface
-func newIPAM(c *Client) *ipams {
-	return &ipams{c, blockReaderWriter{c}}
+// NewIPAM returns a new ipamClient, which implements the IPAMInterface.
+// Consumers of the Calico API should not create this directly, but should
+// access IPAM through the main client IPAM accessor.
+func NewIPAM(client bapi.Client, pools PoolAccessorInterface) *ipams {
+	return &ipams{client: client, pools: pools,
+		blockReaderWriter: blockReaderWriter{client: client, pools: pools}}
 }
 
 // ipamClient implements the IPAMInterface
 type ipams struct {
```

Comment:
can we rename this to something else? also the comment seems wrong

Reply:
renamed to ipamClient as per the comment.